src/auth/jwt.strategy.ts

```diff
@@ -0,0 +1,28 @@
+import { Injectable, UnauthorizedException } from '@nestjs/common';
+import { PassportStrategy } from '@nestjs/passport';
+import { ExtractJwt, Strategy } from 'passport-jwt';
+import { Env } from 'src/env/index';
+import { JwtPayload } from './jwt-interface';
+import { RedisService } from '@liaoliaots/nestjs-redis';
+
+@Injectable()
+export class JwtStrategy extends PassportStrategy(Strategy, 'jwt') {
+  constructor(
+    private readonly redisService: RedisService,
+  ) {
+    super({
+      secretOrKey: Env.JWT.AUTH_TOKEN,
+      jwtFromRequest: ExtractJwt.fromAuthHeaderAsBearerToken(),
+    });
+  }
+
+  async validate(payload: JwtPayload) {
+    const { userId } = payload;
+    const client = this.redisService.getClient();
+    const jwtToken = await client.get(`jwt:${userId}`);
+    if (!jwtToken) {
+      throw new UnauthorizedException();
```

Comment:
PullProofAI

There is a potential performance issue in the `validate` method of the `JwtStrategy` class. The current implementation retrieves the JWT token from Redis for every validation request, which can introduce unnecessary overhead.

#### Recommendation
Consider caching the JWT tokens or optimizing the retrieval process to reduce the number of Redis calls and improve performance.

```typescript
async validate(payload: JwtPayload) {
    const { userId } = payload;
    
    // Check if the JWT token is already cached
    const cachedToken = await this.redisService.getClient().get(`jwt:${userId}`);
    
    if (!cachedToken) {
        throw new UnauthorizedException();
    }
    
    return payload;
}
```

This optimization will help reduce the load on the Redis server and improve the overall performance of the authentication process.


<!-- This is an auto-generated comment by PullProofAI -->